_docs/Qabel-Protocol-Box.md

```diff
@@ -0,0 +1,578 @@
+last_change_by: UUID, // ID of the device that made the last change
+spec_version: INT,  // version of the VOLUME spec
+					// increment if migrations are needed
+					// and/or the files are incompatible between versions
+version: {version: KEY, time: LONG}, // metadata version and time of change, time should not be trusted
+shared: // description of all shares
+{ shares* },
+files: // list of files in this folder
+[ file* ]
+folders: // list of folders in this folder
+[ folder* ]
+externals: // list of external shares in this folder
+[ external* ]
+}
+```
+
+Note that folders that are not "index" do not have the "shared"-key, as all information about shares in a VOLUME are stored in "index".
+
+The version is a SHA-256 hash built with the following rule:
+```
+version(0) = SHA-256(0x00 || device-id)
+version(n) = SHA-256(0x01 || version(n-1) || device-id)
+```
+
+The device-id is unique for each client and a 128-bit value.
+
+### Shares
+
+The index is the path to the metadata file of the share.
+
+```
+{
+read: [KEY] // List of public keys of identities that this folder is read only shared with
+},
+```
+
+### File Metadata [FM]
+
+A file only has its own FM, if it is shared in a single file share.
+The FM is stored in the path referenced as "meta" in the file object and encrypted
+with a new symmetric key
+
+```
+{
+name: STR, // filename
+spec_version: INT,  // version of the VOLUME spec
+size: LONG, // uncompressed file size
+mtime: LONG, // modification time as seconds since epoch
+key: KEY, // symmetric key for the block
+block: STR // path to the block without the prefix \<root\>/blocks/
+}
+```
+
+### Objects
+
+File:
+
+```
+{
+name: STR, // object name,
+size: LONG, // uncompressed file size
+mtime: LONG, // modification time as seconds since epoch
+meta: {ref: STR, // path to the FM, if it exists
+       key: KEY} // symmetric key for the FM
+key: KEY, // symmetric key for the block
+block: STR // path to the block without the prefix \<root\>/blocks/
+},
+```
+
+Folder:
+
+```
+{
+name: STR // object name,
+key: KEY // symmetric directory key
+ref: STR // ref of the metadata file that contains information about the folder
+},
+```
+
+External:
+
+```
+{
+name: STR, // object name,
+key: KEY // symmetric directory key
+owner: STR, // public key of the owner of that VOLUME
+url: STR // URL to the metadata file that contains information about the folder
+},
+```
+
+### dk - Directory Key
+The directory key is stored in the directory object of the parent folder, the index
+file DM is encrypted with the public key of the owner.
+
+### fk - File Key
+File keys are stored in the directory metadata file
+
+### Qabel Identities
+Identities have a public key **pub** and a private key **priv**
+
+### Device ID
+Each client device has a unique ID which is a random generated UUID **devID**
+
+### Path to VOLUME/\<index\>
+\<index\> is calculated from the sha256 of the owners private key. Take the first 128bit of it
```

Comment:
I'm not sure if it is a good idea to use the private key here, since this leaks information of about the private key, even if we use a trap door function...
Any opinions @L-Henke @schulze 
Wouldn't the public key do the job as well?